pmd-core/src/main/java/net/sourceforge/pmd/lang/rule/RuleViolationFactory.java

```diff
@@ -34,8 +44,43 @@
      * @param args
      *            arguments to embed in the rule violation message
      */
-    void addViolation(RuleContext ruleContext, Rule rule, @Nullable Node node, String message, Object[] args);
+    @Deprecated
+    default void addViolation(RuleContext ruleContext, Rule rule, @NonNull Node node, @NonNull String message, Object[] args) {
+        addViolation(ruleContext, rule, node, message, node.getBeginLine(), node.getEndLine(), args);
+    }
+
+
+    @Deprecated
+    default void addViolation(RuleContext ruleContext, Rule rule, @NonNull Node node, @NonNull String message, int beginLine, int endLine, @NonNull Object[] args) {
+        Objects.requireNonNull(node);
+        Objects.requireNonNull(message);
+        Objects.requireNonNull(args);
+        RuleViolation rv = formatViolation(rule, node, ruleContext.getSourceCodeFilename(), message, args);
```

Comment:
`ruleContext.getSourceCodeFilename()` only returns the simple name. We need to use here `String.valueOf(ruleContext.getSourceCodeFile())`. I'll fix that on pmd/7.0.x and add a unit test (hopefully not forgetting that there is also windows...)